frontend/src/StoryDisplays/StoriesList.jsx

```diff
@@ -0,0 +1,104 @@
+import { isPast } from 'date-fns';
+
+class StoryNode {
+  constructor(data, prev = null, next = null) {
+    this.data = data;
+    this.prev = prev;
+    this.next = next;
+  }
+}
+
+class StoriesList {
+  constructor() {
+    this.head = new StoryNode(null);
+    this.tail = new StoryNode(null);
+    this.head.next = this.tail;
+    this.tail.prev = this.head;
+    this.size = 0;
+  }
+
+  insertStory(data) {
+    if (this.size === 0) {
+      this.insertAtEnd(data);
+      return;
+    }
+
+    let currentNode = this.head.next;
+    while (currentNode !== this.tail) {
```

Comment:
this is good, although in industry its typically best practice to avoid using while loops since errors in implementation could cause a never ending loop. Obviously not a blocker, but if we finish early, it would be great to see if we can shift the logic here to utilize an iterative solution (for loop) or if you're feeling very daring, a recursive one. This may require some investigation on your part if this sounds like something you're interested in doing